demo/server/src/index.ts

```diff
@@ -0,0 +1,75 @@
+'use strict';
+/**
+ * Module dependencies.
+*/
+import app from './app';
+import * as http from 'http';
+
+/**
+ * Get port from environment and store in Express.
+ */
+const port = normalizePort(process.env.PORT || 3000);
+app.set('port',port);
+
+/**
+ * Create HTTP server.
+ */
+var server = http.createServer(app);
+
+/**
+ * Listen on provided port,on all network interfaces.
+ */
+server.listen(port, onListening);
+server.on('error',onError);
+
+/**
+ * Normalize a port into a number,string,or false.
+ */
+function normalizePort(val : any): number|string|boolean {
+  let port = parseInt(val,10);
+  
+  if(isNaN(port)){
+    //name pipe
+    return val;
+  }
+  
+  if(port >= 0){
+    return port;
+  }
+  
+  return false;
+}
+
+/**
+ * Event listener for HTTP server "error" event.
+ */
+function onError(error: any) {
+  if (error.syscall != 'listen') {
+    throw error;
+  }
+  
+  var bind = typeof port === 'string' ? 'Pipe ' + port : 'Port ' + port;
+  
+  //handle specific listen errors with friendly messages
+  switch(error.code) {
+    case 'EACCES':
+      console.error(bind + 'requires elevated privileges');
```

Comment:
I'd rather have template strings instead of concat if possible. i.e. `${bind} requires elevated privileges`
This concat for instance is missing the space before `requires`, and that's a kind of error that template strings solve nicely.

Also we'll need to replace `console.error` with the logger module later